src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginTest.groovy

```diff
@@ -329,31 +338,57 @@ class ScalafixPluginTest extends Specification {
-//        applyScalafixPlugin(subproject)
-//
-//        then:
-//        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
-//        task.configFile.get().asFile.path == "${subproject.projectDir}/.scalafix.conf"
-//    }
-//
-//    def 'scalafix uses the .scalafix config file from the root project as the file is not present in the subproject'() {
-//        setup:
-//        Project subproject = ProjectBuilder.builder().withName('the-subproject')
-//                .withParent(project).build()
-//
-//        when:
-//        applyScalafixPlugin(subproject)
-//
-//        then:
-//        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
-//        task.configFile.get().asFile.path == "${project.projectDir}/.scalafix.conf"
-//    }
+    def 'scalafix uses the .scalafix config file provided via extension'() {
+        given:
+        Project subproject = ProjectBuilder.builder().withName('the-subproject')
+                .withParent(project).build()
+        subproject.projectDir.mkdir()
+        File subprojectScalafixConf = new File(subproject.projectDir, '.scalafix.conf')
+        subprojectScalafixConf.write 'rules = [Foo, Bar]'
+        File extensionScalafixConf = new File(subproject.projectDir, '.test-scalafix.conf')
+        extensionScalafixConf.write 'rules = [Foo, Bar]'
+        applyScalafixPlugin(subproject, false, '', extensionScalafixConf)
+
+        when:
+        subproject.evaluate()
+
+        then:
+        println subproject.tasks
+        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
+        task.configFile.get().asFile.path == "${subproject.projectDir}/.test-scalafix.conf"
+    }
+
+    def 'scalafix uses the .scalafix config file from the subproject if it has not been provided via extension'() {
+        given:
+        Project subproject = ProjectBuilder.builder().withName('the-subproject')
+                .withParent(project).build()
+        subproject.projectDir.mkdir()
+        File subprojectScalafixConf = new File(subproject.projectDir, '.scalafix.conf')
+        subprojectScalafixConf.write 'rules = [Foo, Bar]'
+        applyScalafixPlugin(subproject, false, '', null)
+
+        when:
+        subproject.evaluate()
+
+        then:
+        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
+        task.configFile.get().asFile.path == "${subproject.projectDir}/.scalafix.conf"
+    }
+
+    def 'scalafix uses the .scalafix config file from the root project as the file is not present in the subproject and\
+ it is not specified in the extension'() {
+        given:
+        Project subproject = ProjectBuilder.builder().withParent(project).withName('the-subproject').build()
+        scalafixConf.delete()
+        applyScalafixPlugin(subproject, false, '', null)
+
+        when:
+        subproject.evaluate()
+
+        then:
+        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
+        !task.configFile.isPresent()
```

Comment:
I believe the scenario described in the description does not reflect the implementation. Looks like the implementation is testing the scenario where the config file is not informed at all (none of the 3 options we provide).